File: September-2025/03_09_2025_gfg.py
class Node:  # Assuming a Node class exists
    def __init__(self, data):
        self.data = data
        self.next = None
        self.prev = None

    def reverseDLL(self, head: 'Node') -> 'Node':
        if head is None or head.next is None:
            return head

        curr = head
        prev = None

        while curr:
            next_node = curr.next

            curr.next = prev
            curr.prev = next_node

            prev = curr
            curr = next_node

        return prev

File: September-2025/03_09_2025_gfg.cpp
#include <bits/stdc++.h>
using namespace std;

/*
class Node {
  public:
    int data;
    Node *next;
    Node *prev;
    Node(int val) {
        data = val;
        next = NULL;
        prev = NULL;
    }
};

*/
class Solution {
  public:
    Node *reverse(Node *head) {
        // code here
        Node* curr = head;
        Node* temp = nullptr;

        while (curr != nullptr) {
            temp = curr->prev;
            curr->prev = curr->next;
            curr->next = temp;
            curr = curr->prev;
        }

        if (temp != nullptr) {
            head = temp->prev;
        }

        return head;
    }
};